### ISSUE TITLE
PayPal Express Transaction ID wrong on recurring orders.

### ISSUE THREAD

**Issue by andrewlimaza** [strangerstudios/paid-memberships-pro#1077]
**Describe the bug**
When a recurring membership level's order is created, the transaction_id value is set to the subscription_id.

This only happens on recurring membership orders with PayPal Express.

**To Reproduce**
1. Create a recurring membership level.
2. Checkout for this level using PayPal Express.
3. Navigate to the recently created order and notice that the transaction_id is the same as the subscription_id.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only Paid Memberships Pro plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Memberlite](https://wordpress.org/themes/memberlite/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by mircobabini** [comment]
Even if [PayPal docs](https://developer.paypal.com/docs/nvp-soap-api/create-recurring-payments-profile-nvp/) says that CreateRecurringPaymentsProfile Response Fields are:
- PROFILEID
- PROFILESTATUS
- TRANSACTIONID

In fact this is not the case. Instead:
- PROFILEID, always set
- PROFILESTATUS, sometimes is missing. When missing, means error.
- TRANSACTIONID, never set

So, at this moment, the only way to get the missing TRANSACTIONID would be:
- collect the PROFILEID (I-xxxxxxx)
- call the method `$morder->Gateway->getRealPaymentTransactionId( $morder )` which was written exactly to recover the real TRANSACTIONID for order set with TRANSACTIONID=PROFILEID (all PayPal Express orders).

**Comment by tictag** [comment]
**MODS ONLY** Further information on this, including screenshots, are available in plus support ticket: https://www.paidmembershipspro.com/forums/topic/bugfix-paypal-express-transaction-v-subscription-ids-in-order

**Comment by mircobabini** [comment]
Update: checking [NVP PayPal official deprecated documentation](https://developer.paypal.com/docs/api-basics/notifications/ipn/IPNandPDTVariables/)...

<img width="634" alt="Screenshot 2021-05-10 at 16 02 42" src="https://user-images.githubusercontent.com/636911/117671649-50e97a80-b1a9-11eb-90ce-71a014968fe3.png">

That line of the doc is buggy. The `initial_payment_txn_id` is always provided on new profile created message, while it's not passed on recurring payments.

This means that we can update the order transaction_id as soon as we receive the `recurring_payment_profile_created` ipn message. 

To sum up:
1) During checkout process, we can use `$morder->Gateway->getRealPaymentTransactionId( $morder )` to get the only payment collected which is the initial payment and save the order using the correct payment id.
2) As soon as we receive the ipn `recurring_payment_profile_created` ipn message, we can make sure the transaction_id was collected during the checkout or force it to update and throw a warning.

**Comment by mircobabini** [comment]
Fixed by https://github.com/strangerstudios/paid-memberships-pro/commit/67573a1fb7659ae5a88045110e840b6e2089e930: as soon as the real transaction_id comes, it's replaced.

**Comment by dparker1005** [comment]
Closing as resolved
